### PR TITLE
Cleanup temporary files on failure

### PR DESCRIPTION
**Pull request by jeremcs**
Fixes https://github.com/iterative/dvc/issues/1675
Generate temporary file information as we already do for from_info and to_info, and clean these temporary files in case of failure or interruption in the main thread.

### CODE REVIEW

File: dvc/remote/local.py [hunk 1]
```diff
@@ -681,16 +691,30 @@ def _process(
...
+            raise
         for f in futures:
             f.result()
 
         return len(chunks)
 
+    def _cleanup_tmp_files(self, remote, tmp_files):
+        for i_info in tmp_infos:
+            logger.error("About to remove file {}".format(i_info))
+            if i_info["scheme"] == "local":
```
Comment (efiop) [inline dvc/remote/local.py]: The distinction should actually be done not based on scheme, but rather on whether or not we are pushing or pulling. For pulling/fetching it should be `self.remove(i_info)` and for pushing it should be `remote.remove(i_info)`

**Comment by jeremcs** [comment]
> @jeremcs So sorry for the delay. Please check out https://dvc.org/doc/user-guide/contributing. Currently tests fail because your code is not formatted with black and not checked with flake8. In the guide mentioned above, there are instructions on how to setup your env to help with that.

@efiop No worries, and thank you for the review ! I saw the feedback of the CI and I plan to work on it some time this week :+1:

**Comment by efiop** [comment]
Hi @jeremcs ! Do you need any help? :slightly_smiling_face:

**Comment by jeremcs** [comment]
> Hi @jeremcs ! Do you need any help? slightly_smiling_face

Sorry for the late reply.
I can make some time to work on this, I'll let you know if I have any question. Thanks.

**Comment by efiop** [comment]
Closing as stale. Please feel free to submit a new PR or reopen this one :)

**Pull request closed without merging**